Kernel/Locking/Mutex.h

```diff
@@ -82,7 +82,8 @@ class Mutex {
     // FIXME: remove this after annihilating Process::m_big_lock
     using BigLockBlockedThreadList = IntrusiveList<&Thread::m_big_lock_blocked_threads_list_node>;
 
-    void block(Thread&, Mode, SpinlockLocker<Spinlock>&, u32);
+    // FIXME: Allow any lock rank.
+    void block(Thread&, Mode, SpinlockLocker<Spinlock<LockRank::None>>&, u32);
```

Comment:
Couldn't we just make this templated already and manually instantiate templates? We already know all the possible LockRanks, as far as I can tell...

Reply:
To be honest I don't want to stretch this PR any further :^) so that would definitely be a direct follow up